Automattic/wp-calypso
# Import in Signup: Validate site is importable

## Summary

When importing a sign during signup, validates the url and checks the importability of the site against `is-site-importable` endpoint. Shows an appropriate error message if the url is incorrect or the site is not importable.

## Tasks

- [x] Input validation UI
- [x] Check site is importable before proceeding to the next step
- [x] Loading state while checking site import-ability
- [x] Styles for notice positioning, card margins, etc
- [ ] [punted until endpoint is updated]`is-site-importable` error message that differentiates between not finding the site and the site not being compatible with importer
- [x] Importer specific url validation (from `client/my-sites/importer/site-importer/site-importer-input-pane.jsx`)

## Testing Plan

1. Visit `/start/import/from-url`
1. Try submitting with a blank url; you should see an error.
1. Try submitting with an invalid url; you should see an error.
1. Try submitting with a valid url that isn't a real site; you should see a loading state as the site is checked, then an error.
1. Try submitting with a url without a path (i.e. `https://me.wixsite.com`); you should see an error.
1. Try submitting with an admin url (i.e. `https://www.wix.com/website/builder`); you should see an error.
1. Try submitting with an actual site url that is importable; you should be able to proceed to the next step.
